src/App.tsx

```diff
@@ -1,23 +1,92 @@
 import "./App.css";
-
+import React, { useState } from 'react';
 import { taskList } from "./taskList";
 
+let taskListCopy = taskList;
+
+// Create a list with additional field to mark deleted task
+  let taskListExt: any[] = [];
+  for (let i=0 ; i < taskListCopy.length; i++){
+    taskListExt.push(
+      {
+        'id'  : taskListCopy[i].id,
+        'name': taskListCopy[i].name,
+        'del' : '' 
+      }
+    )
+  }
+
+
+//
+
 function TaskShow() {
+  let [tasks, setTasks ] = useState(taskListExt);
+  let [newTask, setNewTask] = useState('');
+  
+  const handleSubmit:React.FormEventHandler<HTMLFormElement> = (event) => {
+    
+    event.preventDefault();
+    let newId  = tasks[ tasks.length-1 ].id + 1;
```

Comment:
This is not re-assigned anywhere else, so use `const` instead

Also, there is no guarantee the last element has the highest `id`, so the new `id` might collide with other element's id

Instead, you can use `for` loop to find the highest `id` from the list. Or you can use [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max) too